### PR TITLE
Update code for hybrid Swift 3 / 4 support.

### PR DESCRIPTION
**Pull request by jshier**
This PR adds support for building AFI using Xcode 9 and Swift 3.2 and 4.0. All projects and schemes have been updated to the latest recommended settings as well.

### CODE REVIEW

File: Source/Request+AlamofireImage.swift [hunk 1]
```diff
@@ -204,7 +204,7 @@ extension DataRequest {
         throw AFIError.imageSerializationFailed
     }
 
-    private class var imageScale: CGFloat {
+    public class var imageScale: CGFloat {
```
Comment (jshier) [inline Source/Request+AlamofireImage.swift]: Add to change this as this value is used as a default in a public method, which Swift 4 doesn't like.

File: Source/ImageCache.swift [hunk 2]
```diff
@@ -196,8 +196,11 @@ open class AutoPurgingImageCache: ImageRequestCache {
             if self.currentMemoryUsage > self.memoryCapacity {
                 let bytesToPurge = self.currentMemoryUsage - self.preferredMemoryUsageAfterPurge
 
-
-                var sortedImages = self.cachedImages.map{$1}
+            #if swift(>=4.0)
+                var sortedImages = self.cachedImages.map{ $0.1 }
```
Comment (cnoon) [inline Source/ImageCache.swift]: Space before the `{`.

File: Source/ImageCache.swift [hunk 3]
```diff
@@ -196,8 +196,11 @@ open class AutoPurgingImageCache: ImageRequestCache {
...
+            #else
+                var sortedImages = self.cachedImages.map { $1 }
+            #endif
```
Comment (cnoon) [inline Source/ImageCache.swift]: Newline?

File: Source/Request+AlamofireImage.swift (hunk 1 above)
Comment (cnoon) [inline Source/Request+AlamofireImage.swift]: Makes sense. 👍🏼

File: Source/ImageCache.swift (hunk 2 above)
Comment (jshier) [inline Source/ImageCache.swift]: Addressed in [1073f6](https://github.com/Alamofire/AlamofireImage/pull/269/commits/1c0598c6b1a6cbdf7a0da362f3a45689e51073f6).

File: Source/ImageCache.swift (hunk 3 above)
Comment (jshier) [inline Source/ImageCache.swift]: Addressed in [1073f6](https://github.com/Alamofire/AlamofireImage/pull/269/commits/1c0598c6b1a6cbdf7a0da362f3a45689e51073f6).

**Comment by caffeinum** [comment]
Is this included into CocoaPods revision? I should have replace these manually in the AlamofireImage (3.2.0) downloaded from Pods.

**Pull request merged**